File: reactivando-backend/src/main/java/com/reactivando/futuro/service/EmpresaService.java
package com.reactivando.futuro.service;

import com.reactivando.futuro.dto.EmpresaRequestDTO;
import com.reactivando.futuro.dto.EmpresaResponseDTO;

import java.util.List;

public interface EmpresaService {
    EmpresaResponseDTO obtenerPorId(Long id);
    EmpresaResponseDTO obtenerPorUsuarioId(Long usuarioId);
    EmpresaResponseDTO obtenerPorCorreo(String correo);
    List<EmpresaResponseDTO> obtenerTodas();
    EmpresaResponseDTO actualizarPerfil(Long usuarioId, EmpresaRequestDTO requestDTO);
    com.reactivando.futuro.dto.empresa.UbicacionEmpresaDTO obtenerUbicacion(Long empresaId);
    EmpresaResponseDTO actualizarCoordenadas(Long usuarioId, Double latitud, Double longitud);
}

File: reactivando-frontend/src/services/empresaService.js
import api from './api';

export const empresaService = {
  obtenerPorId: async (id) => {
    const response = await api.get(`/empresas/${id}`);
    return response.data;
  },

  obtenerPorUsuarioId: async (usuarioId) => {
    const response = await api.get(`/empresas/usuario/${usuarioId}`);
    return response.data;
  },

  obtenerUbicacion: async (id) => {
    const response = await api.get(`/empresas/${id}/ubicacion`);
    return response.data;
  },

  obtenerTodas: async () => {
    const response = await api.get('/empresas');
    return response.data;
  },

  actualizarPerfil: async (usuarioId, empresaData) => {
    const response = await api.put(`/empresas/usuario/${usuarioId}`, empresaData);
    return response.data;
  },

  actualizarCoordenadas: async (usuarioId, latitud, longitud) => {
    const response = await api.put(`/empresas/usuario/${usuarioId}/coordenadas?latitud=${latitud}&longitud=${longitud}`);
    return response.data;
  },
};
